lib/ws/ws.js

```diff
@@ -9,13 +9,86 @@ var factory = require('./factory');
 var authMethod = 'auth';
 
 /**
+ * race takes an array of promises and is resolved with the first promise
+ * that resolves, or rejects. After the first resolve or rejection, all
+ * future resolutions and rejections will be discarded.
+ * @param {Promise[]} promises
+ * @return {Promise}
+ */
+function race(promises) {
```

Comment:
Currently the Beam client is written so that it is agnostic and works with any A+ promise implementation, of which there are many on the Js world. There are fewer who are strict ES6 promise implementations, and RSVP which we use on the frontend is not one. Changing it here would introduce a breaking change for all client consumers, which I'd rather avoid doing.